client/src/api/block_builder/input_selection/automatic.rs

```diff
@@ -218,33 +218,19 @@ impl<'a> ClientBlockBuilder<'a> {
                         Ok(r) => r,
                         // for these errors, just try again in the next round with more addresses which might have more
                         // outputs.
-                        Err(err @ crate::Error::InsufficientAmount { .. }) => {
-                            cached_error.replace(err);
-                            continue;
-                        }
-                        Err(err @ crate::Error::NotEnoughNativeTokens { .. }) => {
-                            cached_error.replace(err);
-                            continue;
-                        }
-                        // Native tokens left, but no balance for the storage deposit for a remainder.
-                        Err(err @ crate::Error::NoBalanceForNativeTokenRemainder) => {
-                            cached_error.replace(err);
-                            continue;
-                        }
-                        // Currently too many inputs, by scanning for more inputs, we might find some with more amount.
-                        Err(err @ crate::Error::ConsolidationRequired { .. }) => {
-                            cached_error.replace(err);
+                        Err(err @ InputSelectionError::InsufficientAmount { .. }) => {
+                            cached_error.replace(Error::from(err));
                             continue;
                         }
                         // Not enough balance for a remainder.
-                        Err(crate::Error::Block(block_error)) => match block_error {
+                        Err(InputSelectionError::Block(block_error)) => match block_error {
                             iota_types::block::Error::InvalidStorageDepositAmount { .. } => {
-                                cached_error.replace(crate::Error::Block(block_error));
+                                cached_error.replace(Error::from(InputSelectionError::Block(block_error)));
                                 continue;
                             }
                             _ => return Err(block_error.into()),
                         },
-                        Err(e) => return Err(e),
+                        Err(e) => return Err(e)?,
```

Comment:
InsufficientAmount, InsufficientNativeTokenAmount and NoAvailableInputsProvided should still not return an error here?

Reply:
Added back

Reply:
Can we event get `InsufficientAmount` there? And what about `NoAvailableInputsProvided`?